File: kotlin/src/test/kotlin/cn/staynoob/trap/kotlin/basic/operator/Fixture.java
package cn.staynoob.trap.kotlin.basic.operator;

public class Fixture {
    static class Operand {
        int value;

        public Operand(int value) {
            this.value = value;
        }

        public int plus(Operand other) {
            return this.value + other.value;
        }

        public int subtract(Operand other) {
            return this.value - other.value;
        }
    }
}

File: kotlin/src/test/kotlin/cn/staynoob/trap/kotlin/basic/object/Fixture.kt
package cn.staynoob.trap.kotlin.basic.`object`

object ObjectExpression {
    val foo = "foo"
    fun fn() = "fn"
}

class ClassWithCompanion {
    companion object {
        val foo = "foo"
        fun fn() = "fn"
    }
}

class ClassWithNamingCompanion {
    companion object WithName{
        val foo = "foo"
        fun fn() = "fn"
    }
}

class UserFactory private constructor(val account: String) {
    companion object {
        fun createUserByName(name: String) = UserFactory(name)
        fun createUserByPhone(phone: String) = UserFactory(phone)
        fun createUserByEmail(email: String) = UserFactory(email)
    }
}